### DISCUSSION TITLE
Consider moving sizeToBytes from validator to formatter.

### DISCUSSION THREAD

**Discussion by Eseperio** [yiisoft/yii2#19526] [Ideas]
This method is often useful when working with systems that implement sizes with digits+text notation. Code is available as private method within FileValidator, while it could be integrated onto formatter. 

https://github.com/yiisoft/yii2/blob/161526cd41beee62d0910333c3bd96034cf73186/framework/validators/FileValidator.php#L377

**Comment by WinterSilence** [comment]
Same function `ini_parse_quantity()` will be added in PHP 8.2

**Comment by samdark** [comment]
That makes sense overall. Feel free to make a pull request.

**Comment by mspirkov** [comment]
I think this is a good idea. But in my opinion, it's better to move this method to `BaseStringHelper`. Here's the PR: https://github.com/yiisoft/yii2/pull/20550

**Reply by Eseperio** [reply]
Size to bytes is clearly a format change, so i do not see the link with a helper for strings.

**Reply by mspirkov** [reply]
This is an auxiliary function that does not depend on any external settings. It seems to me that the formatter contains primarily those methods that depend on any external settings.

**Reply by mspirkov** [reply]
I also believe that you should not use the global Yii class too often, as it can make testing more difficult.

**Reply by mspirkov** [reply]
@samdark @terabytesoftw @Eseperio 
The pull request has been merged. Is there anything else we'd like to discuss?

**Reply by mspirkov** [reply]
I would like to make sure that everything is ok before adding this feature to `yiisoft/strings` (https://github.com/yiisoft/yii2/pull/20550#issuecomment-3349263690)

**Comment by mspirkov** [comment]
I also added this feature to `yiisoft/strings` (https://github.com/yiisoft/strings/commit/c8770364932bab9cea4f25f6a77f75eb37991eec)
